### PR TITLE
remove `depth` and `max_depth` from GTTA and walk validation

### LINKED ISSUE #19: remove the `belowMaxDepth` validation criterion

### LINKED ISSUE #31: remove depth from tipselection
- [ ] entrypoint

- [ ] config

- [ ] maxDepth

### PR DESCRIPTION
**Pull request by alon-e**
Fixes #31 
Fixes #19 

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

- Compilation and unit tests

### CODE REVIEW

File: src/main/java/com/iota/iri/service/tipselection/impl/TipSelectorImpl.java [hunk 1]
```diff
@@ -60,21 +60,20 @@ public TipSelectorImpl(Tangle tangle,
...
+    public List<Hash> getTransactionsToApprove(Optional<Hash> reference) throws Exception {
 
         //preparation
-        Hash entryPoint = entryPointSelector.getEntryPoint(depth);
+        Hash entryPoint = entryPointSelector.getEntryPoint();
         UnIterableMap<HashId, Integer> rating = ratingCalculator.calculate(entryPoint);
 
         //random walk
         List<Hash> tips = new LinkedList<>();
-        WalkValidator walkValidator = new WalkValidatorImpl(tangle, ledgerValidator, config);
```
Comment (Wollac) [inline src/main/java/com/iota/iri/service/tipselection/impl/TipSelectorImpl.java:77]: The private field `config` is no longer needed.

File: src/main/java/com/iota/iri/service/tipselection/impl/EntryPointSelectorGenesisImpl.java [hunk 2]
```diff
@@ -16,6 +17,11 @@ public EntryPointSelectorGenesisImpl() {
 
     @Override
     public Hash getEntryPoint(int depth) {
+        throw new NotImplementedException("Not supported in CLIRI");
```
Comment (Wollac) [inline src/main/java/com/iota/iri/service/tipselection/impl/EntryPointSelectorGenesisImpl.java]: Independently of cliri, the method is never supported for Genesis EntryPointSelector. Maybe change the exception message.

File: src/main/java/com/iota/iri/service/tipselection/impl/WalkValidatorImpl.java [hunk 3]
```diff
@@ -28,15 +27,13 @@
     private final Tangle tangle;
     private final Logger log = LoggerFactory.getLogger(WalkValidator.class);
     private final LedgerValidator ledgerValidator;
-    private final TipSelConfig config;
```
Comment (Wollac) [inline src/main/java/com/iota/iri/service/tipselection/impl/WalkValidatorImpl.java:31]: Line 4: unused import `com.iota.iri.conf.TipSelConfig`

Comment (Wollac) [inline src/main/java/com/iota/iri/service/tipselection/impl/WalkValidatorImpl.java:31]: Line 4: unused import `com.iota.iri.conf.TipSelConfig`

File: src/test/java/com/iota/iri/service/tipselection/impl/WalkValidatorImplTest.java [hunk 4]
```diff
@@ -29,12 +29,9 @@
     private static final TemporaryFolder dbFolder = new TemporaryFolder();
     private static final TemporaryFolder logFolder = new TemporaryFolder();
     private static Tangle tangle;
-    private TipSelConfig config = new MainnetConfig();
```
Comment (Wollac) [inline src/test/java/com/iota/iri/service/tipselection/impl/WalkValidatorImplTest.java:32]: Unused imports for `TipSelConfig`, `MainnetConfig` and `TransactionValidator`.

**Comment by alon-e** [comment]
@Wollac thanks!
addressed review comments.

**Pull request merged**